lib/task_listener.js

```diff
@@ -212,24 +212,57 @@ export default class TaskListener extends EventEmitter {
     }
   }
 
+  addRunningTask(runningState) {
+    this.runningTasks.push(runningState);
+    this.incrementPending();
+  }
+
+  removeRunningTask(runningState) {
+    let taskIndex = this.runningTasks.findIndex((runningTask) => {
+      return (runningTask.taskId === runningState.taskId) &&
+        (runningTask.runId === runningState.runId)
+    });
+
+    if (taskIndex === -1) {
+      this.runtime.log('[warning] running task removal error', {
+        taskId: runningState.taskId,
+        runId: runningState.runId,
+        err: 'Could not find the task Id in the list of running tasks'
+      });
+      return;
+    }
+
+    this.cleanupRunningState(runningState);
```

Comment:
Good idea.  It's possible that something threw while aquiring the cpu/loopback devices or whatever else that could mean it's not in the list yet as running but still needs things to clean up.